scalafix/rules/src/main/scala/fix/FixBqSaveAsTable.scala

```diff
@@ -11,28 +11,50 @@ class FixBqSaveAsTable extends SemanticRule("FixBqSaveAsTable") {
 
   override def fix(implicit doc: SemanticDocument): Patch = {
     doc.tree.collect {
-      case a @ Term.Apply(fun, head :: tail) =>
+      case a @ Term.Apply(fun, params) =>
         if (fun.symbol.normalized.toString.contains(methodName)) {
           fun match {
             case Term.Select(qual, name) =>
               name match {
                 case Term.Name("saveAvroAsBigQuery") if expectedType(qual, scoll) =>
-                  // the rest of args should be named because the parameter order has changed
-                  if (
-                    tail.exists(!_.toString.contains("=")) ||
-                    // `avroSchema` doesn't exist in the list of the new method
-                    tail.exists(_.toString.contains("avroSchema"))
-                  ) {
-                    Patch.empty // not possible to fix, leave it to `LintBqSaveAsTable`
-                  } else {
-                    val headParam = if (head.toString.contains("=")) {
-                      s"table = Table.Ref(${head.toString.split("=").last.trim})"
-                    } else {
-                      s"Table.Ref($head)"
-                    }
-                    val allArgs = (headParam :: tail).mkString(", ")
-                    Patch.replaceTree(a, s"$qual.saveAsBigQueryTable($allArgs)")
-                  }
+                  val paramsUpdated =
+                    params
+                      .zipWithIndex
+                      .foldLeft(List[Object]()) { case (accumulator, (param, index)) =>
+                        accumulator :+ (
+                          index match {
+                            // table is always the first param and without default value
+                            case 0 =>
+                              param match {
+                                case Term.Assign((_, value)) =>
+                                  s"table = Table.Ref(${value})"
+                                case _ =>
+                                  s"Table.Ref($param)"
+                              }
+                            case _ =>
+                              param match {
+                                case Term.Assign((name, value)) =>
+                                  // parameter name has changes from `avroSchema` to `schema`
+                                  if (name.toString == "avroSchema") {
+                                    s"schema = toTableSchema($value)"
+                                  } else {
+                                    param
+                                  }
+                                case _ =>
+                                  // if not a named param, `avroSchema` param should come second
+                                  if (index == 1) {
+                                    s"toTableSchema($param)"
+                                  } else {
+                                    param
+                                  }
```

Comment:
Here you are mixing Tree and String
We should work with quasiquotes to return trees only.

Reply:
can you explain what can go wrong doing so?